crates/miden-tx/src/errors/mod.rs

```diff
@@ -27,6 +27,8 @@ use thiserror::Error;
 
 #[derive(Debug, Error)]
 pub enum NoteCheckerError {
+    #[error("invalid input note count {0})")]
+    InvalidInputNoteCount(u16),
```

Comment:
I would like to get rid of this so that I can do `From<NoteCheckerError> for TransactionExecutorError` (see below.

But I'm not sure it makes sense to remove this variant either.

Reply:
One approach is to move this variant into TransactionExecutorError. It makes the code a bit cleaner but also feels wrong to have a checker-level variant pushed down to TransactionExecutorError.